test/reductions/thomasvandoren/test/TestScanCounts.future

```diff
@@ -0,0 +1,3 @@
+bug: scan constructor fails to resolve
+
+See the error message in the .bad file.
```

Comment:
Note that this is another example of a class inheriting from an uninstantiated generic class, as came up on email this week and was briefly discussed in yesterday's team meeting.

All that said, I would expect this to work simply by using 'counts scan ...' rather than needing to define scanCounts.  Does it not?